Fabric.Identity.API/Management/UserLoginManager.cs

```diff
@@ -65,6 +62,22 @@ private void SetNamePropertiesFromClaims(User user)
             }
         }
 
+        private string GetUserName(User user)
+        {
+            var userName = string.Empty;
+            var upn = user.Claims.FirstOrDefault(c => c.Type == FabricIdentityConstants.PublicClaimTypes.UserPrincipalName)?.Value;
+            var name = user.Claims.FirstOrDefault(c => c.Type == JwtClaimTypes.Name)?.Value;

```

Comment:
Minor optimization - could you change the `else if` below to an `else` and do this lookup for the `Name` claim in that?